### PR TITLE
MolGAN Torch Model

### PR DESCRIPTION
**Pull request by shreyasvinaya**
## Description

Add molgan torch model to deepchem

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/molgan.py [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from deepchem.feat.molecule_featurizers.molgan_featurizer import GraphMatrix
+from deepchem.models.torch_models import WGANModel
+from deepchem.models.torch_models.layers import MolGANEncoderLayer
+
+
+class Discriminator(nn.Module):
+    """A discriminator for the MolGAN model."""
```
Comment (rbharath) [inline deepchem/models/torch_models/molgan.py:12]: Add usage example

File: deepchem/models/torch_models/molgan.py [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+
+    def __init__(self,
+                 dropout_rate: float,
+                 units: List = [(128, 64), 64],
+                 edges: int = 5,
+                 device: Optional[torch.device] = torch.device('cpu')):
```
Comment (rbharath) [inline deepchem/models/torch_models/molgan.py:18]: docstring

File: deepchem/models/torch_models/molgan.py [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+        # Define the dense layers
+        self.dense1 = nn.Linear(units[1], 128)
+        self.dropout1 = nn.Dropout(dropout_rate)
+        self.dense2 = nn.Linear(128, 64)
+        self.dropout2 = nn.Dropout(dropout_rate)
+        self.dense3 = nn.Linear(64, 1)
+
+    def forward(
+        self,
+        inputs: List[torch.Tensor],
```
Comment (rbharath) [inline deepchem/models/torch_models/molgan.py:37]: docstring

File: deepchem/models/torch_models/molgan.py (hunk 2 above)
Comment (shreyasvinaya) [reply deepchem/models/torch_models/molgan.py]: fixed

File: deepchem/models/torch_models/molgan.py (hunk 3 above)
Comment (shreyasvinaya) [reply deepchem/models/torch_models/molgan.py]: fixed

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**